tests/context.cpp

```diff
@@ -18,6 +18,26 @@ TEST_CASE("context create, close and destroy", "[context]")
     CHECK(NULL == (void *) context);
 }
 
+TEST_CASE("context shutdown", "[context]")
+{
+    zmq::context_t context;
+    context.shutdown();
+    CHECK(NULL != (void *) context);
+    context.close();
+    CHECK(NULL == (void *) context);
+}
+
+TEST_CASE("context shutdown often", "[context]")
+{
+    zmq::context_t context;
+    context.shutdown();
+    context.shutdown();
+    context.shutdown();
```

Comment:
Calling it three times is a bit confusing. To check that it can be called more than once, two calls were sufficient. To check if it can still be called when all sockets have been closed or so, a different test would be needed.